src/transformers/tokenization_utils_base.py

```diff
@@ -0,0 +1,1775 @@
+        Returns:
+            :obj:`CharSpan`:
+                Span of characters in the original string.
+
+                :obj:`CharSpan` are NamedTuple with:
+
+                - start: index of the first character in the original string
+                - end: index of the character following the last character in the original string
+        """
+
+        if not self._encodings:
+            raise ValueError("token_to_chars() is not available when using Python based tokenizers")
+        if token_index is not None:
+            batch_index = batch_or_token_index
+        else:
+            batch_index = 0
+            token_index = batch_or_token_index
+        return CharSpan(*(self._encodings[batch_index].token_to_chars(token_index)))
+
+    def char_to_token(self, batch_or_char_index: int, char_index: Optional[int] = None) -> int:
+        """
+        Get the index of the token in the encoded output comprising a character
+        in the original string for a sequence of the batch.
+
+        Can be called as:
+
+        - ``self.char_to_token(char_index)`` if batch size is 1
+        - ``self.char_to_token(batch_index, char_index)`` if batch size is greater or equal to 1
+
+        This method is particularly suited when the input sequences are provided as
+        pre-tokenized sequences (i.e. words are defined by the user). In this case it allows
+        to easily associate encoded tokens with provided tokenized words.
+
+        Args:
+            batch_or_char_index (:obj:`int`):
+                Index of the sequence in the batch. If the batch only comprise one sequence,
+                this can be the index of the word in the sequence
+            char_index (:obj:`int`, `optional`):
+                If a batch index is provided in `batch_or_token_index`, this can be the index
+                of the word in the sequence.
+
+
+        Returns:
+            :obj:`int`: Index of the token.
+        """
+
+        if not self._encodings:
+            raise ValueError("char_to_token() is not available when using Python based tokenizers")
+        if char_index is not None:
+            batch_index = batch_or_char_index
+        else:
+            batch_index = 0
+            char_index = batch_or_char_index
+        return self._encodings[batch_index].char_to_token(char_index)
+
+    def word_to_chars(self, batch_or_word_index: int, word_index: Optional[int] = None) -> CharSpan:
+        """
+        Get the character span in the original string corresponding to given word in a sequence
+        of the batch.
+
+        Character spans are returned as a CharSpan NamedTuple with:
+
+        - start: index of the first character in the original string
+        - end: index of the character following the last character in the original string
+
+        Can be called as:
+
+        - ``self.word_to_chars(word_index)`` if batch size is 1
+        - ``self.word_to_chars(batch_index, word_index)`` if batch size is greater or equal to 1
```

Comment:
When I look at this example: `word_index` is mandatory in both cased and `batch_index` is optionally. However, the type hints suggest, that `word_index` is optional (see line 377). I think this should be corrected then :)